src/context/mapStateSlice.ts

```diff
@@ -0,0 +1,39 @@
+import { Map, Set } from 'immutable';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { RootState } from './store';
+import { LayerType } from '../config/types';
+
+interface LayersSet extends Set<LayerType> {}
+interface MapState extends Map<string, LayersSet | Date> {}
+
+const initialState: MapState = Map({
+  layers: Set<LayerType>(),
+  currentDate: new Date(),
+});
+
+export const mapStateSlice = createSlice({
+  name: 'mapState',
+  initialState,
+  reducers: {
+    addLayer: (state, { payload }: PayloadAction<LayerType>) =>
+      state.update('layers', layers => (layers as LayersSet).add(payload)),
+
+    removeLayer: (state, { payload }: PayloadAction<LayerType>) =>
+      state.update('layers', layers => (layers as LayersSet).delete(payload)),
+
+    updateDate: (state, { payload }: PayloadAction<Date>) =>
+      state.set('currentDate', payload),
+  },
+});
+
+// Getters
+export const selectLayers = (state: RootState) =>
```

Comment:
The name `selectLayers` sounds weird.
Do you mean something like `getAvailableLayers` or `selectedLayers`?

Reply:
@ericboucher 
I named it like that as it is used like that `const layers = useSelector(selectLayers);` I tried to follow the convention on [redux documentation](https://react-redux.js.org/api/hooks#useselector-examples)

Another option can be `layersSelector`?

Reply:
Yeah I think I'd prefer that :)